File: app/src/main/java/Model/Notifications/Notification.java
package Model.Notifications;

import com.google.firebase.Timestamp;

public class Notification implements Comparable<Notification>{
    private Timestamp Date;
    private int type;

    public Timestamp getDate() {
        return Date;
    }

    public void setDate(Timestamp date) {
        Date = date;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    @Override
    public int compareTo(Notification o) {
        return o.getDate().compareTo(getDate());
    }
}

File: app/src/main/java/com/whiteside/insta/model/Notification.kt
package com.whiteside.insta.model

import com.google.firebase.Timestamp

open class Notification(val type: Int)

class LikeNotification(
    var postID: String,
    var UID: String,
    var date: Timestamp,
) : Notification(1)

class FriendRequestNotification(
    var UID: String? = null,
    var date: Timestamp,
) : Notification(2)